plugins/tekton/src/utils/tekton-utils.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  ClusterObjects,
+  ObjectsByEntityResponse,
+} from '@backstage/plugin-kubernetes-common';
+import { pluralize } from '@patternfly/react-core';
+import { PipelineRunGVK, TaskRunGVK } from '../models';
+import {
+  ComputedStatus,
+  SucceedConditionReason,
+  TaskStatus,
+} from '../types/computedStatus';
+import { PipelineRunKind } from '../types/pipelineRun';
+import { ClusterErrors, TektonResponseData } from '../types/types';
+import {
+  pipelineRunFilterReducer,
+  pipelineRunStatus,
+} from './pipeline-filter-reducer';
+
+export const getClusters = (k8sObjects: ObjectsByEntityResponse) => {
+  const clusters: string[] = k8sObjects.items.map(
+    (item: ClusterObjects) => item.cluster.name,
+  );
+  const errors: ClusterErrors[] = k8sObjects.items.map(
+    (item: ClusterObjects) => item.errors,
+  );
+  return { clusters, errors };
+};
+
+const isTektonResource = (kind: string) =>
+  [PipelineRunGVK.kind, TaskRunGVK.kind].includes(kind);
+
+const getResourceType = (kind: string) => {
+  switch (kind) {
+    case PipelineRunGVK.kind:
+      return 'pipelineruns';
+    case TaskRunGVK.kind:
+      return 'taskruns';
+    default:
+      return '';
+  }
+};
+
+export const getTektonResources = (
+  cluster: number,
+  k8sObjects: ObjectsByEntityResponse,
+) =>
+  k8sObjects.items?.[cluster]?.resources?.reduce(
+    (acc: TektonResponseData, res: any) => {
+      if (
+        res.type !== 'customresources' ||
+        (res.type === 'customresources' && res.resources.length === 0)
+      ) {
+        return acc;
+      }
+      const customResKind = res.resources[0].kind;
+      return {
+        ...acc,
+        ...(isTektonResource(customResKind) && {
+          [getResourceType(customResKind)]: {
+            data: res.resources,
+          },
+        }),
+      };
+    },
+    {},
+  );
+
+export const totalPipelineRunTasks = (pipelinerun: PipelineRunKind): number => {
+  if (!pipelinerun?.status?.pipelineSpec) {
+    return 0;
+  }
+  const totalTasks = (pipelinerun.status.pipelineSpec?.tasks || []).length;
+  const finallyTasks =
+    (pipelinerun.status.pipelineSpec?.finally || []).length ?? 0;
+  return totalTasks + finallyTasks;
+};
+
+export const updateTaskStatus = (pipelinerun: PipelineRunKind): TaskStatus => {
+  const skippedTaskLength = pipelinerun?.status?.skippedTasks?.length || 0;
+  const taskStatus: TaskStatus = {
+    PipelineNotStarted: 0,
+    Pending: 0,
+    Running: 0,
+    Succeeded: 0,
+    Failed: 0,
+    Cancelled: 0,
+    Skipped: skippedTaskLength,
+  };
+  if (!pipelinerun?.status?.taskRuns) {
+    return taskStatus;
+  }
+  const plrTasks = Object.keys(pipelinerun.status.taskRuns);
```

Comment:
we shouldn't be relying on this deprecated field anymore `pipelinerun.status.taskRuns` instead use the taskruns retrieved as a custom resource.